### ISSUE TITLE
Models creation fails if /tmp is a symlink

### ISSUE THREAD

**Issue by mehdiym** [volatiletech/sqlboiler#577]
### What version of SQLBoiler are you using (`sqlboiler --version`)?
SQLBoiler v3.4.0

### If this happened at generation time what was the full SQLBoiler command you used to generate your models? (if not applicable leave blank)
sqlboiler mysql

### What is the output of the command above with the `-d` flag added to it? (Provided you are comfortable sharing this, it contains a blueprint of your schema)
`unable to fetch table column info (friend_requests): Error 1: Can't create/write to file '/tmp/#sql_3a5_0.MAI' (Errcode: 13 "Permission denied")
Error: unable to initialize tables: unable to fetch table data: driver (/home/myo/go/bin/sqlboiler-mysql) exited non-zero: exit status 1`

### Please provide a relevant database schema so we can replicate your issue (Provided you are comfortable sharing this)
Set your /tmp directory as a symlink to somewhere else.

### Further information. What did you do, what did you expect?
I fixed it by make the /tmp directory plain again.

**Comment by aarondl** [comment]
It sounds like there was a problem with your permissions not with the symlink itself. Can you verify?

**Comment by mehdiym** [comment]
Here are the symlink rights:
```lrwxrwxrwx   1 root root    10 Aug  1 07:47 tmp -> /home/tmp/```
And in the tmp dir in the home directory:
```drwxrwxrwt 17 root root 65536 Aug  1 07:32 tmp```

**Comment by aarondl** [comment]
Right, why does it belong to root? What user is using sqlboiler? What is the permissions of /home/tmp? (which is a very strange directory btw)

**Comment by mehdiym** [comment]
The `/tmp` dir always belongs to root, isn't it?
The permissions of the `/home/tmp` directory are the second ones I wrote just above.
As you can see above, that folder has the sticky bit (`t` flag), as any tmp folder should have. That flag allows any user to create anything in that folder and remove only his own files.
So with a symlink it should work like with a plain directory. Other programs on my computer have no problem with my symlink.

**Comment by aarondl** [comment]
When looking at the code, I don't see anything out of the ordinary. It uses standard library functions to create a temp file. Why that would fail is beyond me. I'm not interested in solving this bug myself. I'd accept a PR to make it work though.

**Comment by mehdiym** [comment]
No worries, as I've changed my tmp dir back to a regular directory, I don't have a problem anymore. Thanks for your time on this.
